.github/workflows/e2e-test.yaml

```diff
@@ -19,14 +22,41 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v3
 
-      - name: Setup Kind cluster
-        id: setup-kind
+      - name: Setup Kind cluster1
+        id: setup-kind1
         uses: ./.github/actions/kind-create
+        with:
+          kind_cluster_name: kind1
+          kind_k8s_version: v1.24.13
+          kubeconfig_dir_path: tests/e2e/kubeconfigs
+
+      # TODO: ZookeeperCluster create time out always on the second kind cluster
+      # - name: Setup Kind cluster2
+      #   id: setup-kind2
+      #   uses: ./.github/actions/kind-create
+      #   with:
+      #     kind_cluster_name: kind2
+      #     kind_k8s_version:  v1.25.9
+      #     kubeconfig_dir_path: tests/e2e/kubeconfigs
 
-      - name: run tests
+      - name: Run E2E tests
         env:
-           KUBECONFIG: ${{ steps.setup-kind.outputs.kubeconfig }}
+           TEST_STRATEGY: version
+           KUBECONFIG_DIR: kubeconfigs
+           MAX_TIMEOUT: "1h"
         run: |
           go work init
           go work use -r .
           make test-e2e
+
+      - name: Publish Test Results
+        uses: EnricoMi/publish-unit-test-result-action/composite@master
+        if: always()
```

Comment:
Question: Do we need this if?

Reply:
I dont know. I copy pasted from the documentation.

Reply:
Looked into it, yeah, we need it, for reference: https://stackoverflow.com/questions/58858429/how-to-run-a-github-actions-step-even-if-the-previous-step-fails-while-still-f#58859404